### PR TITLE
Add docs for fast link times, esp. about WASM_BIGINT

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: site/source/docs/optimizing/Optimizing-Code.rst [hunk 1]
```diff
@@ -41,8 +41,45 @@ Compiling source files to object files works as you'd expect in a native build s
...
+is ok to link with those flags even if the source files were compiled with a
+different optimization level.
+
+``-O0`` will do no optimization work at link time. ``-O1`` will do very minimal
+optimizations, and does not have the assertions that ``-O0`` does by default,
+so it can be useful for a build that links very quickly but also is reasonably
+fast. (Of course, for a final release build, it is usually worth linking with
+something like ``-O3 --closure 1`` for full optimizations.)
+
+In some cases Emscripten can do no work after the LLVM linker (``wasm-ld``) runs.
```
Comment (sbc100) [inline site/source/docs/optimizing/Optimizing-Code.rst:57]: How about `In some cases Emscripten can avoid modifying the wasm binary that is produced by the linker...`?

File: site/source/docs/optimizing/Optimizing-Code.rst [hunk 2]
```diff
@@ -41,8 +41,45 @@ Compiling source files to object files works as you'd expect in a native build s
...
+That will give you the fastest possible link times. All Emscripten does
+in such a build is to generate the JavaScript support code, while leaving the
+WebAssembly output from the linker unmodified.
+
+Specifically, as of Emscripten 2.0.7, if you build with either ``-O0`` or
+``-O1`` then the only thing Emscripten does on the wasm file is to legalize it.
```
Comment (sbc100) [inline site/source/docs/optimizing/Optimizing-Code.rst:63]: `does on...` -> `needs to do to the wasm file is legalize it.`

File: site/source/docs/optimizing/Optimizing-Code.rst [hunk 3]
```diff
@@ -41,8 +41,45 @@ Compiling source files to object files works as you'd expect in a native build s
...
+You can avoid even that by enabling BigInt integration, as when using BigInts
```
Comment (sbc100) [inline site/source/docs/optimizing/Optimizing-Code.rst:64]: `You can avoid even that by`   -> `This can avoided by enabling BigInt integration which renders legalization unnecessary`.

**Review by sbc100 — APPROVED** [review]

File: site/source/docs/optimizing/Optimizing-Code.rst [hunk 4]
```diff
@@ -41,8 +41,45 @@ Compiling source files to object files works as you'd expect in a native build s
...
+
+``-O0`` will do no optimization work at link time. ``-O1`` will do very minimal
+optimizations, and does not have the assertions that ``-O0`` does by default,
+so it can be useful for a build that links very quickly but also is reasonably
+fast. (Of course, for a final release build, it is usually worth linking with
+something like ``-O3 --closure 1`` for full optimizations.)
+
+In some cases Emscripten can do no work after the LLVM linker (``wasm-ld``) runs.
+That will give you the fastest possible link times. All Emscripten does
+in such a build is to generate the JavaScript support code, while leaving the
```
Comment (RReverser) [inline site/source/docs/optimizing/Optimizing-Code.rst:59]: Not 100% sure, but probably
```suggestion
in such a build is generation of the JavaScript support code, while leaving the
```

File: site/source/docs/optimizing/Optimizing-Code.rst [hunk 5]
```diff
@@ -41,8 +41,45 @@ Compiling source files to object files works as you'd expect in a native build s
...
+
+.. code-block:: bash
+
+  emcc -s WASM_BIGINT -s ERROR_ON_WASM_CHANGES_AFTER_LINK
+
+``ERROR_ON_WASM_CHANGES_AFTER_LINK`` will, as the name implies, show an error
+during link if Emscripten must perform changes to the Wasm. If you remove that
+``-s WASM_BIGINT``, it will tell you that legalization forces it to change the
+wasm. You will also get an error if you build with ``-O2`` or above, as then it
+will run the Binaryen optimizer.
```
Comment (RReverser) [inline site/source/docs/optimizing/Optimizing-Code.rst:82]:
```suggestion
will attempt to run the Binaryen optimizer.
```
?

File: site/source/docs/optimizing/Optimizing-Code.rst [hunk 6]
```diff
@@ -41,8 +41,45 @@ Compiling source files to object files works as you'd expect in a native build s
...
+Link Times
+==========
+
+To skip extra optimization work at link time, link with ``-O0`` or ``-O1``. It
+is ok to link with those flags even if the source files were compiled with a
+different optimization level.
+
+``-O0`` will do no optimization work at link time. ``-O1`` will do very minimal
+optimizations, and does not have the assertions that ``-O0`` does by default,
+so it can be useful for a build that links very quickly but also is reasonably
```
Comment (RReverser) [inline site/source/docs/optimizing/Optimizing-Code.rst:53]: I feel like "very quickly but also is reasonable fast" is a bit confusing. It would be good to rephrase it to make it clear that in the 2nd part it's the resulting code that will be reasonably fast, too, and not the build as the process itself.

File: site/source/docs/optimizing/Optimizing-Code.rst (hunk 4 above)
Comment (kripken) [reply site/source/docs/optimizing/Optimizing-Code.rst:59]: I think `is generate` (`All Emscripten does in such a build is generate the JavaScript support code`) is better than both of those, but others here have better English than me...

Comment (kripken) [reply site/source/docs/optimizing/Optimizing-Code.rst:59]: (but thanks, yeah, my original here was not great...)

File: site/source/docs/optimizing/Optimizing-Code.rst (hunk 5 above)
Comment (kripken) [reply site/source/docs/optimizing/Optimizing-Code.rst:82]: "Attempt" may be confusing, but I agree this could be better. I'll change to `You will also get an error if you build with ``-O2`` or above, as the Binaryen optimizer would normally be run.` -?

File: site/source/docs/optimizing/Optimizing-Code.rst (hunk 6 above)
Comment (kripken) [reply site/source/docs/optimizing/Optimizing-Code.rst:53]: Agreed, thanks, will change to `but also runs reasonably fast`.

**Comment by kripken** [comment]
Thanks for the comments, should be addressed now.

**Review by sbc100 — APPROVED** [review]
